Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -159,4 +159,57 @@ describe('VirtualizedList', () => {
     );
     expect(component).toMatchSnapshot();
   });
+
+  it('returns the viewableItems correctly in the onViewableItemsChanged callback after changing the data', () => {
+    const ITEM_HEIGHT = 600;
+    let data = [{key: 'i1'}, {key: 'i2'}, {key: 'i3'}];
+    const scrollEvent = {
+      timeStamp: 1000,
+      nativeEvent: {
+        contentOffset: {y: 0, x: 0},
+        layoutMeasurement: {width: 300, height: 600},
+        contentSize: {width: 300, height: data.length * ITEM_HEIGHT},
+        zoomScale: 1,
+        contentInset: {right: 0, top: 0, left: 0, bottom: 0},
+      },
+    };
+    const onViewableItemsChanged = jest.fn();
+    const props = {
+      data,
+      renderItem: ({item}) => <item value={item.key} />,
+      getItem: (data, index) => data[index],
+      getItemCount: data => data.length,
+      getItemLayout: (data, index) => ({
+        length: ITEM_HEIGHT,
+        offset: ITEM_HEIGHT * index,
+        index,
+      }),
+      onViewableItemsChanged,
+    };
+
+    const component = ReactTestRenderer.create(<VirtualizedList {...props} />);
+
+    const instance = component.getInstance();
+
+    instance._onScrollBeginDrag(scrollEvent);
+    instance._onScroll(scrollEvent);
+
+    expect(onViewableItemsChanged).toHaveBeenCalledTimes(1);
+    expect(onViewableItemsChanged).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        viewableItems: [expect.objectContaining({isViewable: true, key: 'i1'})],
+      }),
+    );
+    data = [{key: 'i4'}, ...data];
+    component.update(<VirtualizedList {...props} data={data} />);
+
+    instance._onScroll(scrollEvent);
```

Comment:
Hmm, perhaps you should change the scroll event offset so it's more realistic? I could imagine adding some optimization that filters out scroll events that are identical and this test breaking.